src/drive/ducks/upload/index.js

```diff
@@ -120,3 +121,62 @@ export const getUploadQueue = state => state[SLUG].queue
 export const getProcessed = state =>
   getUploadQueue(state).filter(f => f.status !== PENDING)
 export const getSuccessful = state => getLoaded(getUploadQueue(state))
+
+// DnD helpers for folder upload
+export const canHandleFolders = evt => {
+  if (!evt.dataTransfer) return false
+  const dt = evt.dataTransfer
+  return dt.items && dt.items.length && dt.items[0].webkitGetAsEntry != null
+}
+
+const getFileFromEntry = async entry => {
+  let result
+  await new Promise(resolve => {
+    entry.file(file => {
+      result = file
+      resolve()
```

Comment:
You should put `results` in the function scope instead of using a closure and `resolve(results)` to return the promised value, shouldn't you?

Reply:
Or it can be rewritten like this :

```
return await new Promise(resolve => entry.file(resolve))
```

A bit extreme but you get the idea.

Reply:
Actually `return new Promise(resolve => entry.file(resolve))` would be the same, and more concise.

Reply:
Good point!